core/src/main/java/com/linecorp/armeria/server/cors/CorsService.java

```diff
@@ -108,6 +109,58 @@ protected HttpObject filter(HttpObject obj) {
         };
     }
 
+    /**
+     * This is a non CORS specification feature which enables the setting of preflight
+     * response headers that might be required by intermediaries.
+     *
+     * @param headers the {@link HttpHeaders} to which the preflight headers should be added.
+     */
+    void setPreflightHeaders(final CorsPolicy policy, final HttpHeaders headers) {
+        requireNonNull(policy, "policy");
+        for (Map.Entry<AsciiString, String> entry : policy.preflightResponseHeaders()) {
+            headers.add(entry.getKey(), entry.getValue());
+        }
+    }
+
+    void setCorsAllowCredentials(final CorsPolicy policy, final HttpHeaders headers) {
+        requireNonNull(policy, "policy");
+        if (policy.isCredentialsAllowed() &&
+            !headers.get(HttpHeaderNames.ACCESS_CONTROL_ALLOW_ORIGIN).equals(CorsService.ANY_ORIGIN)) {
```

Comment:
`headers.get(...)` could return `null`? How about flipping the equals caller and callee?